REF Extract getDefaultRoleID for add participant

mattwire opened the pull request:
Overview
----------------------------------------
Extract function to get default role ID if it's not defined.

Before
----------------------------------------
Custom query was being run whether or not it was required.

After
----------------------------------------
Custom query replaced by API call and only run when it's actually needed (ie. we don't already have a role ID).

Technical Details
----------------------------------------

Comments
----------------------------------------

pradpnayak commented on CRM/Event/Form/Registration.php:
```diff
@@ -857,9 +840,7 @@ public static function addParticipant(&$form, $contactID) {
-      'role_id' => CRM_Utils_Array::value('participant_role_id',
-        $params, $roleID
-      ),
+      'role_id' => self::getDefaultRoleID($params),
```
How about 
```
'role_id' => self::getDefaultRoleID(
  CRM_Utils_Array::value('participant_role_id', $params)
),

```
`private static function getDefaultRoleID($roleID) {`

monishdeb replied:
or how about: 
```php
...
'role_id' => CRM_Utils_Array::value(
   'participant_role_id',
    $params,
    civicrm_api3('OptionValue', 'getvalue', [
        'return' => "value",
        'option_group_id' => "participant_role",
        'options' => ['limit' => 1],
      ])
  ),
...
```
:)

pradpnayak replied:
I agree with creating a new function for getDefaultRoleID() since it can be re-used elsewhere but i think we should not pass whole params array to that function

eileenmcnaughton replied:
I agree - the only reason params is being passed is to check whether we need to look up the default role id - I think that should be done outside the function ie 

$roleId = CRM_Utils_Array::value('participant_role_id', $params) ? : self::getDefaultRoleID();

mattwire commented:
test this please

eileenmcnaughton commented on CRM/Event/Form/Registration.php:
```diff
@@ -911,6 +892,26 @@ public static function addParticipant(&$form, $contactID) {
+    $roleID = CRM_Utils_Array::value('participant_role_id', $params);
+    if (!$roleID) {
+      $roleID = civicrm_api3('OptionValue', 'getvalue', [
+        'return' => "value",
```
is the is_default=1 not required? Is is_active implicit?

mattwire commented:
@monishdeb @pradpnayak @eileenmcnaughton Thanks for the review.  I've taken the best of each and updated the PR.

I've added in is_active (that should have been there).  Also I've sorted by is_default though they're all NULL on my db and it wasn't previously checking for the is_default flag but it makes sense that it does...

seamuslee001 commented:
Seems like this has general agreement and has passed unit tests merging
